### ISSUE TITLE
Feature Request: Extension points to enable timeline as drag target for creating new events

### ISSUE THREAD

**Issue by foufee** [namespace-ee/react-calendar-timeline#60]
For a project I'm currently working on, I would like to be able to drag something on to the timeline, and then have a callback with the dropped event, and the time at which the drop happened.  
Additionally I will need to be able to drag something onto the items as well, and have a callback fired.
It is going to be used for a scheduling application that creates a schedule that associates several resources together, akin to scheduling a meeting room, and then dragging contacts onto the event.

I'm happy to attempt to make the changes myself, but was wondering if this feature is something you would be interested in once complete, and if you were available to provide guidance if required?

**Comment by mariusandra** [comment]
Hey, that would be a cool feature, as long as it's disabled by default of course :).

For pointers on how to do it, look at http://interactjs.io/ . It's the library we currently use for all the resize/move/touch interactions. It also supports dropzones.

Good luck!

**Comment by DanielRuf** [comment]
Also works with react-dnd without any issues. You need drag elements and wrap the target / drop element around the timeline. Get the offset of the dragged item over the timeline, calculate the start time using this information and push a new item to the state on drop.

**Comment by foufee** [comment]
In my clone of the repo is an initial attempt of this. I've added a bunch of callbacks on the Timeline props object that will allow a user to be notified when dropping a draggable on any of the groups, or when dropping a draggable from outside the timeline onto any of the horizontal lines in the timeline.
Also I added contextMenu, and selection callbacks for the timeline HLines and the groups.

The demo index.html shows most of the features off (a bit crudely at the moment though).

I feel a bit more cleaning up and refactoring could be done (a lot of the code is similar between HorizontalLine and Group), but I'm on a tight deadline and need to progress with other stuff at the moment (it's close enough at the moment for me)

I would be interested in any feedback you have.

Cheers
